### PR TITLE
CSP-772 Adds PUT Attendance

### PR DESCRIPTION
**Pull request by SawNawDfE**

### CODE REVIEW

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs [hunk 1]
```diff
@@ -23,4 +25,17 @@ public async Task<IActionResult> GetCalendarEvents([FromHeader(Name = Constants.
         var response = await _mediator.Send(new GetCalendarEventsQuery(requestedByMemberId), cancellationToken);
         return Ok(response);
     }
+
+    [HttpPut("{calendarEventId}/attendance")]
```
Comment (MarkFCain) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs:29]: needs coverage

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs [hunk 2]
```diff
@@ -23,4 +25,17 @@ public async Task<IActionResult> GetCalendarEvents([FromHeader(Name = Constants.
...
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    public async Task<IActionResult> PutAttendance(
+    Guid calendarEventId,
+    [FromHeader(Name = "X-RequestedByMemberId")] Guid requestedByMemberId,
```
Comment (MarkFCain) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs:33]: can you change this inline "X-..." to refer to a constant?

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/Attendances/Commands/PutAttendance/PutAttendanceCommandHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿using AutoFixture.NUnit3;
+using FluentAssertions;
+using MediatR;
+using Moq;
+using SFA.DAS.ApprenticeAan.Application.Attendances.Commands.PutAttendance;
+using SFA.DAS.ApprenticeAan.Application.Infrastructure;
+using SFA.DAS.Testing.AutoFixture;
+
+namespace SFA.DAS.ApprenticeAan.Application.UnitTests.Attendances.Commands.PutAttendance
```
Comment (DevenKShah) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/Attendances/Commands/PutAttendance/PutAttendanceCommandHandlerTests.cs:9]: file level namespace

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs (hunk 1 above)
Comment (SawNawDfE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/CalendarEventsController.cs:29]: Added

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/Attendances/Commands/PutAttendance/PutAttendanceCommandHandlerTests.cs (hunk 3 above)
Comment (SawNawDfE) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/Attendances/Commands/PutAttendance/PutAttendanceCommandHandlerTests.cs]: Done

**Pull request merged**